src/classes.rs

```diff
@@ -149,14 +149,21 @@ impl<'a> Iterator for ByteClassRepresentatives<'a> {
 /// This particular representation only permits contiguous ranges of bytes to
 /// be in the same equivalence class, which means that we can never discover
 /// the true minimal set of equivalence classes.
-#[derive(Debug)]
-pub struct ByteClassBuilder(Vec<bool>);
+pub struct ByteClassBuilder([bool; 256]);
+
+impl fmt::Debug for ByteClassBuilder {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_tuple("ByteClassBuilder")
+            .field(&&self.0[..])
+            .finish()
+    }
+}
 
 impl ByteClassBuilder {
     /// Create a new builder of byte classes where all bytes are part of the
     /// same equivalence class.
     pub fn new() -> ByteClassBuilder {
-        ByteClassBuilder(vec![false; 256])
+        ByteClassBuilder([false; 256])
```

Comment:
Is there a particular motivation for this? It seems like any performance overhead of the additional allocation would be negligible compared to the additional work being done to build an AC automaton.

Reply:
Yeah it doesn't have enough of an effect to be noticeable. Just figured that the added impls wouldn't need to be touched again after so it wouldn't be any burden. I can remove the change if you want.

Alternatively it the `[S; 256]` might be used enough to warrant an abstraction which can be used everywhere in the crate. It does incur a rather unfortunate `Clone` impl however... :/ https://github.com/Marwes/aho-corasick/commit/322e19b853069b28b4392feb631cc8a14e5eb1a2

Reply:
Yeah I'd probably prefer just switching back to `Vec<bool>`. I definitely don't think the generic structure is worth it.

Reply:
👍 Removed those commits.

(I can remove the unsafe with a macro and copy paste https://github.com/Marwes/aho-corasick/commit/87c3602845b61f3223a9c56c00c4fc80831f7679 if that makes `ByteMap` more palatable).

Reply:
Hmm, I guess the `[S; 256]` constraint is mostly self-inflicted. Only `ByteClasses` used an array beforehand so `ByteMap` could just use a `Vec` and `ByteClasses` could use an array still.

Reply:
I mostly just think that generics are overkill for this. :-) Sorry!

Reply:
Aww :( . Everything should be fixed in the PR though, the `ByteMap` stuff was all in a separate branch.